vectordb_bench/backend/clients/pgvector/config.py

```diff
@@ -46,48 +79,142 @@ def parse_metric_fun_str(self) -> str:
             return "max_inner_product"
         return "cosine_distance"
 
+    @abstractmethod
+    def index_param(self) -> PgVectorIndexParam:
+        ...
+
+    @abstractmethod
+    def search_param(self) -> PgVectorSearchParam:
+        ...
+
+    @abstractmethod
+    def session_param(self) -> PgVectorSessionCommands:
+        ...
+
+    @staticmethod
+    def _optionally_build_with_options(with_options: Mapping[str, Any]) -> Sequence[dict[str, Any]]:
+        """Walk through mappings, creating a List of {key1 = value} pairs. That will be used to build a where clause"""
+        options = []
+        for option_name, value in with_options.items():
+            if value is not None:
+                options.append(
+                    {
+                        "option_name": option_name,
+                        "val": str(value),
+                    }
+                )
+        return options
+
+    @staticmethod
+    def _optionally_build_set_options(
+        set_mapping: Mapping[str, Any]
+<<<<<<< HEAD
+    ) -> Sequence[dict[str, Any]]:
+        """Walk through options, creating 'SET 'key1 = "value1";' commands"""
+        session_options = []
+=======
+    ) -> Sequence[dict[str, Any]]:
+        session_options = []
+>>>>>>> 1a4f510 (Remove import of psycopg from config.)
+        for setting_name, value in set_mapping.items():
+            if value:
+                session_options.append(
```

Comment:
@jkatz need resolve.

Reply:
🤦 That's embarrassing. I'll fix that on the next pass.

Reply:
For the record sake - I had previously resolved a merge conflict, but for some reason it left the merge flags in. That is likely my misread from VSCode. The next commit fixes this.